src/main/java/mate/academy/mybookshop/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto, UserEntity user) {
+        Set<CartItemEntity> cartItems = cartItemRepository
+                .findCartItemEntitiesByShoppingCartId(user.getId());
+        if (cartItems.isEmpty()) {
+            throw new EmptyCartException("Cannot create order without items.");
+        }
+
+        OrderEntity order = orderMapper.toEntity(requestDto);
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+
+        order.setTotal(calculateTotal(cartItems));
+        order.setShippingAddress(orderMapper.toEntity(requestDto).getShippingAddress());
+
+        Set<OrderItemEntity> orderItems = new HashSet<>();
+        for (CartItemEntity cartItem : cartItems) {
+            OrderItemEntity orderItem = new OrderItemEntity();
+            orderItem.setOrder(order);
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice().multiply(
+                    new BigDecimal(cartItem.getQuantity())));
+            orderItems.add(orderItem);
+        }
+        order.setOrderItems(orderItems);
+
+        orderRepository.save(order);
+        cartItemRepository.deleteAll(cartItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderResponseDto> getAllOrders(UserEntity user, Pageable pageable) {
+        return orderRepository.getAllByUserId(user.getId(), pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getOrderItems(
+            UserEntity user, Long orderId, Pageable pageable) {
+        return orderItemRepository.findByOrderIdAndUserId(user.getId(), orderId, pageable).stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItem(UserEntity user, Long orderId, Long orderItemId) {
+        return orderItemRepository.findByOrderIdAndItemIdAndUserId(
+                        user.getId(), orderId, orderItemId)
+                .map(orderItemMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Order item not found in the specified order."));
+    }
+
+    @Override
+    public void changeStatus(Long orderId, UpdateOrderRequestDto requestDto) {
+        OrderEntity order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Order not found for the given user."));
```

Comment:
you are not looking up order by the user
```suggestion
                .orElseThrow(() -> new EntityNotFoundException(
                        "Order with id: " + orderId + " does not exist"));
```

Reply:
fixed